Remove data from 'wrong' table

fredden opened the pull request:
When the `backend_type` of an attribute changes (eg, from `varchar` to `text`), Magento will not clean up / migrate the data from the old table automatically. Extensions should do this as part of changing the `backend_type`, but not all extensions follow best practices.
This pull request updates the SQL used to identify orphaned data to also detect data in the 'wrong' table.

sprankhub commented:
Thanks for your PR! You are absolutely right and this should do what you intend to do :) I am hesitant to merge it right now, because this now deletes more than the description says, right? Would you mind extending the description of the command?

If you would add the command to the README as well when you are at it, that would be awesome as well, but we can also do this "later".

fredden commented:
Technically the description does match the behaviour if one reads "parent entry in eav_attribute" as matching the backend_type too. I've updated the description to be more explicit about what it will remove, and added this command to the list of commands in the README.

sprankhub commented:
:heart_eyes: Thanks!
